hdk-rust/tests/integration_test.rs

```diff
@@ -302,10 +303,67 @@ fn example_valid_entry_address() -> Address {
     Address::from("QmefcRdCAXM2kbgLW2pMzqWhUvKSDvwfFSVkvmwKvBQBHd")
 }
 
+#[allow(dead_code)]
+fn wait_for_action(
+    signal_receiver: &SignalReceiver,
+    expected_signal: ConsistencyEvent,
+    max_tries: i8,
+    up_till: Option<usize>,
+) -> Result<ConsistencyEvent, HolochainError> {
+    if max_tries > 0 {
+        signal_receiver
+            //THIS FIRST PART IS JUST FOR OPTIMIZATION PURPOSES WHILE FINDING A GOOD BALANCE OF SEPERATION
+            //============================================================================================
+            .try_iter()
+            //skip the first
+            .skip(up_till.unwrap_or(0))
+            //only iterate over the 10
+            .take(10)
+            //make sure we filter only over the signals
+            .filter(|recv| match recv {
+                Signal::Consistency(_) => true,
+                _ => false,
+            })
+            //==============================================================================================
+            .filter_map(|recv| match recv {
+                Signal::Consistency(cons) => {
+                    let content = cons.clone().event();
+                    let event: ConsistencyEvent =
+                        serde_json::from_str(&content).expect("Could not convert serde");
+                    match event {
+                        ConsistencyEvent::Hold(_) => Some(event),
+                        ConsistencyEvent::AddLink(_) => Some(event),
+                        ConsistencyEvent::UpdateEntry(_, _) => Some(event),
+                        ConsistencyEvent::RemoveEntry(_, _) => Some(event),
+                        ConsistencyEvent::RemoveLink(_) => Some(event),
+                        _ => None,
```

Comment:
Why are you filtering out certain signals? Wouldn't it be OK for the user to specify any signal to wait for?

Reply:
The purpose is just to reduce the sample size that the find checks against, maybe a panic would fair better than a non